libs/src/services/creatorNode/index.js

```diff
@@ -417,6 +430,55 @@ class CreatorNode {
     this.authToken = null
   }
 
+  /**
+   * Gets and returns the clock values across the replica set for the wallet in userStateManager.
+   * @returns {Object[]} Array of objects with the structure:
+   *
+   * {
+   *  type: 'primary' or 'secondary',
+   *  endpoint: <Content Node endpoint>,
+   *  clockValue: clock value (should be an integer) or null
+   * }
+   *
+   * 'clockValue' may be null if the request to fetch the clock value fails
+   */
+  async getClockValuesFromReplicaSet () {
+    const user = this.userStateManager.getCurrentUser()
+    if (!user || !user.creator_node_endpoint) {
+      console.error('No user or Content Node endpoint found')
+      return
+    }
+
+    const primary = CreatorNode.getPrimary(user.creator_node_endpoint)
+    const replicaSet = CreatorNode.getEndpoints(user.creator_node_endpoint)
+
+    const clockValueRequest = async (endpoint) => {
```

Comment:
why is `clockValueRequest` embedded in the function `getClockValuesFromReplicaSet`?

Reply:
if it's in order to specify 'primary' or 'secondary' in the returned values, is there a cleaner way to do that?

Reply:
I figure that `clockValueRequest` was only used in this scope, so I kept it in this fn. this was a wrapper fn around `getClockValue` to not throw errors

Reply:
i see the reasoning and it makes but I think exposing it might be valuable - in fact that could be a useful function to build on if we ever need more granular access from within mad-dog. i believe there is also an actual cost to nesting the functions, like the entire function `clockValueRequest` is recreated each time which can hurt performance (not too important in this case i think but worth noting). maybe something like `getClockValue(endpoint, wallet)` as its own function within this file?